File: DongPoTea.Web/src/types/order.ts
import { Address } from './common';

export interface Order {
  id: string;
  orderNumber: string;
  customerId?: string;
  customerEmail: string;
  subtotal: number;
  tax: number;
  shippingCost: number;
  total: number;
  currency: string;
  status: OrderStatus;
  paymentStatus: PaymentStatus;
  paymentMethod?: string;
  shippingAddress: Address;
  billingAddress: Address;
  notes?: string;
  orderItems: OrderItem[];
  createdAt: string;
  updatedAt?: string;
}

export interface OrderItem {
  id: string;
  productId: string;
  productName: string;
  productSKU: string;
  quantity: number;
  unitPrice: number;
  total: number;
}

export interface CreateOrderRequest {
  customerInfo: {
    email: string;
    firstName: string;
    lastName: string;
    phone?: string;
  };
  shippingAddress: Address;
  billingAddress: Address;
  paymentMethodId: string;
}

export type OrderStatus = 
  | 'Pending'
  | 'Processing'
  | 'Shipped'
  | 'Delivered'
  | 'Cancelled';

export type PaymentStatus = 
  | 'Pending'
  | 'Paid'
  | 'Failed'
  | 'Refunded';

File: DongPoTea.Core/Entities/Order.cs
namespace DongPoTea.Core.Entities;

public class Order : BaseEntity
{
    public string OrderNumber { get; set; } = string.Empty;
    public Guid? CustomerId { get; set; }
    public string CustomerEmail { get; set; } = string.Empty;
    public decimal SubTotal { get; set; }
    public decimal Tax { get; set; } = 0;
    public decimal ShippingCost { get; set; } = 0;
    public decimal Total { get; set; }
    public string Currency { get; set; } = "USD";
    public string Status { get; set; } = "Pending";
    public string PaymentStatus { get; set; } = "Pending";
    public string? PaymentMethod { get; set; }
    public string? StripePaymentIntentId { get; set; }
    public string? ShippingAddressJson { get; set; }
    public string? BillingAddressJson { get; set; }
    public string? Notes { get; set; }
    
    public Customer? Customer { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}